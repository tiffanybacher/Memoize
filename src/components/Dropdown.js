import React, { Component } from 'react';

class Dropdown extends Component {
  constructor(props) {
    super(props);
  }

  handleMouseLeave = () => {
    this.props.hideDropdown();
  }

  goToAllFlashcards = () => {
    this.props.hideMainFlashcard();
    this.props.showAllFlashcards();
    this.props.hideDropdown();
  }

  goToMissedFlashcards = () => {
   this.props.hideMainFlashcard();
   this.props.showMissedFlashcards();
   this.props.hideDropdown();
  }

  startOver = () => {
    console.log('start over')
    // create pop up to ask if user is sure
    // remove saved cards from local storage
    // remove card selection from local storage
    // reset all state defaults
  }

  render() {
    return (
      <article className="Dropdown">
        <ul onMouseLeave={this.handleMouseLeave}>
          <li onClick={this.goToAllFlashcards}>All Flashcards</li>
          <li onClick={this.goToMissedFlashcards}>Flashcards for Review</li>
          <li onClick={this.startOver}>Start Over</li>
        </ul>
      </article>
    );
  }
}

export default Dropdown;