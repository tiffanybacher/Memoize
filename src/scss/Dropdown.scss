.Dropdown {
  background-color: white;
  border-radius: $rounded-border;
  box-shadow: -3px 3px 1px rgba(0, 0, 0, 0.2);
  color: $pink;
  font-size: 1.3em;
  font-weight: 600;
  position: fixed;
  right: 15px;
  top: 55px;

  ul {
    list-style: none;
    margin: 0;
    padding: 0;

    li {
      padding: 10px 25px 10px;
      
      &:first-child {
        margin-top: 10px;
      }

      &:last-child {
        margin-bottom: 10px;
      }

      &:hover {
        background-color: $pink;
        color: white;
        cursor: pointer;
      }
    }
  }
}