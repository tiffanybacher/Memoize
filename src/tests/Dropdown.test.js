import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Dropdown from '../components/Dropdown';

const mockShowDropdown = false;
const mockHideDropdown = jest.fn();

describe('Dropdown', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Dropdown
        showDropdown={mockShowDropdown}
        hideDropdown={mockHideDropdown} />
    );
  });
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should hide dropdown menu on mouseleave', () => {
  wrapper.instance().handleMouseLeave();

  expect(mockHideDropdown).toBeCalled();
  });
});