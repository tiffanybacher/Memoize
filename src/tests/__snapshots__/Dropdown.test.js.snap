// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match snapshot 1`] = `
<article
  className="Dropdown"
>
  <ul
    onMouseLeave={[Function]}
  >
    <li>
      All Flashcards
    </li>
    <li>
      Missed Flashcards
    </li>
    <li>
      Start Over
    </li>
  </ul>
</article>
`;
